superset/viz.py

```diff
@@ -1097,6 +1097,7 @@ def query_obj(self):
             raise Exception(_('Pick a metric!'))
         d['metrics'] = [self.form_data.get('metric')]
         self.form_data['metric'] = metric
+        d['row_limit'] = None
```

Comment:
Rather than hardcoding that we dont set row_limit we will expose this to user.
https://github.com/Guavus/incubator-superset/blob/fix/MIQ-2130-fix-big-chart-query/superset/assets/src/explore/visTypes.jsx#L1407
https://github.com/Guavus/incubator-superset/blob/fix/MIQ-2130-fix-big-chart-query/superset/assets/src/explore/visTypes.jsx#L1435

we will add a control `['row_limit']`

Add a `None` or `-1` here https://github.com/Guavus/incubator-superset/blob/fix/MIQ-2130-fix-big-chart-query/superset/assets/src/explore/controls.jsx#L98

Override row_limit choices or dafult can be achieved by
https://github.com/Guavus/incubator-superset/blob/fix/MIQ-2130-fix-big-chart-query/superset/assets/src/explore/visTypes.jsx#L289-L298

This should get you going. In case you still need further assistance let me know

Reply:
Thanks for the references, it helped a lot.

On selection of "-1", LIMIT is removed from the query. I dropped the idea of "None" as the options are getting validated if they are an integer or not; hence raises an error if "None" is there in options.